### DISCUSSION TITLE
What's the specific purpose of different keys?

### DISCUSSION THREAD

**Discussion by yarnauy** [Chia-Network/chia-blockchain#1732] [Q&A]
There are several kinds of keys in Chia protocol including 
1. plot key
2. pool key
3. farmer key
4. some other keys?

What's the specific purpose of these keys?

**Comment by hoffmang9** [comment]
The wiki is your best resource:

https://github.com/Chia-Network/chia-blockchain/wiki/Chia-Keys-Architecture

https://github.com/Chia-Network/chia-blockchain/wiki/Chia-Keys-Management

**Reply by yarnauy** [reply]
Thanks a lot !!
Btw, Does Chia support both  "Farm to public key" and "Farm to contract address"  now?
